File: Website_Freelancer_BE/src/main/java/com/freelancer/utils/EndPoint.java
package com.freelancer.utils;

public class EndPoint {
    public class Unique {
        public static class Skill {
            public static final String NAME = "unique/skills/name";
        }

        public static class Major {
            public static final String NAME = "unique/major/name";
        }

        public static class Language {
            public static final String NAME = "unique/languages/name";
            public static final String ISO = "unique/languages/iso";
        }

        public static class User {
            public static final String EMAIL = "unique/users/email";
            public static final String PHONE = "unique/users/phone";
        }

        public static class Role {
            public static final String NAME = "unique/roles/name";
        }
    }

    public class Admin {
        public static final String ME = "admin/me";

        public static class Skill {
            public static final String BASE = "admin/skills";
            public static final String ID = BASE + "/{id}";
            public static final String FILTER = BASE + "/filter";
            public static final String VALIDATION = BASE + "/validation";
            public static final String FORM = ID + "/form";
            public static final String ACTIVE = ID + "/active";
            public static final String SORT = BASE + "/sort";
            public static final String IMPACT = ID + "/impact";
            public static final String INVALID = ID + "/invalid";
        }

        public static class Major {
            public static final String BASE = "admin/majors";
            public static final String ID = BASE + "/{id}";
            public static final String FILTER = BASE + "/filter";
            public static final String SORT = BASE + "/sort";
            public static final String VALIDATION = BASE + "/validation";
            public static final String FORM = ID + "/form";
            public static final String ACTIVE = ID + "/active";
            public static final String IMPACT = ID + "/impact";
            public static final String INVALID = ID + "/invalid";
        }

        public static class Language {
            public static final String BASE = "admin/languages";
            public static final String ID = BASE + "/{id}";
            public static final String FILTER = BASE + "/filter";
            public static final String SORT = BASE + "/sort";
            public static final String VALIDATION = BASE + "/validation";
            public static final String FORM = ID + "/form";
            public static final String ACTIVE = ID + "/active";
            public static final String IMPACT = ID + "/impact";
            public static final String INVALID = ID + "/invalid";
        }

        public static class Role {
            public static final String BASE = "admin/roles";
            public static final String ID = BASE + "/{id}";
            public static final String FILTER = BASE + "/filter";
            public static final String SORT = BASE + "/sort";
            public static final String VALIDATION = BASE + "/validation";
            public static final String FORM = ID + "/form";
            public static final String ACTIVE = ID + "/active";
            public static final String INVALID = ID + "/invalid";
        }

        public static class Client {
            public static final String BASE = "admin/clients";
            public static final String ID = BASE + "/{id}";
            public static final String FILTER = BASE + "/filter";
            public static final String VALIDATION = BASE + "/validation";
            public static final String SORT = BASE + "/sort";
            public static final String FORM = ID + "/form";
        }

        public static class Staff {
            public static final String BASE = "admin/staffs";
            public static final String ID = BASE + "/{id}";
            public static final String FILTER = BASE + "/filter";
            public static final String VALIDATION = BASE + "/validation";
            public static final String SORT = BASE + "/sort";
            public static final String FORM = ID + "/form";
            public static final String INVALID = ID + "/invalid";
            public static final String ACTIVE = ID + "/active";
        }
    }

    public static class Auth {
        public static final String LOGIN = "auth/login";
        public static final String FORGOT = "auth/forgot";
        public static final String REGISTER = "auth/register/{role}";
        public static final String LOGIN_VALIDATION = "auth/login/validation";
        public static final String REGISTER_VALIDATION = "auth/register/validation";
    }

    public static class Job {
        public static final String BASE = "jobs";
        public static final String ID = BASE + "/{id}";
        public static final String FILTER = BASE + "/filter";
        public static final String SORT = BASE + "/sort";
        public static final String POST_PUBLIC = ID + "/post-public";
        public static final String POST_PRIVATE = ID + "/post-private";
        public static final String PUBLIC = ID + "/public";
        public static final String APPLY = ID + "/apply";
        public static final String SELECT_FREELANCER = ID + "/select-freelancer/{applyId}";
        public static final String APPLIES = ID + "/applies";
        public static final String REJECT_APPLY = ID + "/reject-apply/{applyId}";

        public static class Step1 {
            private static final String BASE = Job.BASE + "/step1";
            public static final String VALIDATION = BASE + "/validation";
            public static final String ID = BASE + "/{id}";
        }

        public static class Step2 {
            private static final String BASE = Job.BASE + "/step2";
            public static final String VALIDATION = BASE + "/validation";
            public static final String ID = BASE + "/{id}";
        }

        public static class Step3 {
            private static final String BASE = Job.BASE + "/step3";
            public static final String VALIDATION = BASE + "/validation";
            public static final String ID = BASE + "/{id}";
        }

        public static class Step4 {
            private static final String BASE = Job.BASE + "/step4";
            public static final String VALIDATION = BASE + "/validation";
            public static final String ID = BASE + "/{id}";
        }
    }

    public static class Product {
        public static final String BASE = "products";
        public static final String ID = BASE + "/{id}";
        public static final String ACCEPT = ID + "/accept";
        public static final String REJECT = ID + "/reject";
    }

    public static class List {
        private static final String BASE = "list";
        public static final String MAJOR = BASE + "/majors";
        public static final String LANGUAGE = BASE + "/languages";
        public static final String SKILL = BASE + "/skills";
        public static final String ROLE = BASE + "/roles";
        public static final String PERMISSION = BASE + "/permissions";
    }

    public static class Me {
        public static final String BASE = "me";
        public static final String FULL_INFO = BASE + "/full-info";
        public static final String APPLIES = BASE + "/applies";
        public static final String REVIEWS = BASE + "/reviews";
        public static final String JOBS_IN_PROGRESS = BASE + "/jobs-in-progress";
        public static final String JOBS_COMPLETED = BASE + "/jobs-completed";
    }

    public static class Home {
        private static final String BASE = "home";
        public static final String MAJOR = BASE + "/top-10-most-reputable-freelancers";

    }
}

File: website_freelancer_mobile/api/endpoint.ts
export namespace EndPoint {
    export namespace Admin {
        export const ME = 'admin/me';

        export namespace Skill {
            export const BASE = 'admin/skills';
            export const ID = `${BASE}/{id}`;
            export const FILTER = `${BASE}/filter`;
            export const VALIDATION = `${BASE}/validation`;
            export const FORM = `${ID}/form`;
            export const ACTIVE = `${ID}/active`;
            export const SORT = `${BASE}/sort`;
            export const IMPACT = `${ID}/impact`;
            export const INVALID = `${ID}/invalid`;
        }

        export namespace Major {
            export const BASE = 'admin/majors';
            export const ID = `${BASE}/{id}`;
            export const FILTER = `${BASE}/filter`;
            export const SORT = `${BASE}/sort`;
            export const VALIDATION = `${BASE}/validation`;
            export const FORM = `${ID}/form`;
            export const ACTIVE = `${ID}/active`;
            export const IMPACT = `${ID}/impact`;
            export const INVALID = `${ID}/invalid`;
        }

        export namespace Language {
            export const BASE = 'admin/languages';
            export const ID = `${BASE}/{id}`;
            export const FILTER = `${BASE}/filter`;
            export const SORT = `${BASE}/sort`;
            export const VALIDATION = `${BASE}/validation`;
            export const FORM = `${ID}/form`;
            export const ACTIVE = `${ID}/active`;
            export const IMPACT = `${ID}/impact`;
            export const INVALID = `${ID}/invalid`;
        }

        export namespace Role {
            export const BASE = 'admin/roles';
            export const ID = `${BASE}/{id}`;
            export const FILTER = `${BASE}/filter`;
            export const SORT = `${BASE}/sort`;
            export const VALIDATION = `${BASE}/validation`;
            export const FORM = `${ID}/form`;
            export const ACTIVE = `${ID}/active`;
            export const INVALID = `${ID}/invalid`;
        }

        export namespace Client {
            export const BASE = 'admin/clients';
            export const ID = `${BASE}/{id}`;
            export const FILTER = `${BASE}/filter`;
            export const VALIDATION = `${BASE}/validation`;
            export const SORT = `${BASE}/sort`;
            export const FORM = `${ID}/form`;
        }

        export namespace Staff {
            export const BASE = 'admin/staffs';
            export const ID = `${BASE}/{id}`;
            export const FILTER = `${BASE}/filter`;
            export const VALIDATION = `${BASE}/validation`;
            export const SORT = `${BASE}/sort`;
            export const FORM = `${ID}/form`;
            export const INVALID = `${ID}/invalid`;
            export const ACTIVE = `${ID}/active`;
        }
    }

    export namespace Auth {
        export const LOGIN = 'auth/login';
        export const FORGOT = 'auth/forgot';
        export const REGISTER = 'auth/register/{role}';
        export const LOGIN_VALIDATION = 'auth/login/validation';
        export const REGISTER_VALIDATION = 'auth/register/validation';
    }

    export namespace Job {
        export const BASE = 'jobs';
        export const ID = `${BASE}/{id}`;
        export const FILTER = `${BASE}/filter`;
        export const SORT = `${BASE}/sort`;
        export const POST_PUBLIC = `${ID}/post-public`;
        export const POST_PRIVATE = `${ID}/post-private`;
        export const PUBLIC = `${ID}/public`;

        export namespace Step1 {
            const BASE_STEP = `${Job.BASE}/step1`;
            export const VALIDATION = `${BASE_STEP}/validation`;
            export const ID = `${BASE_STEP}/{id}`;
        }

        export namespace Step2 {
            const BASE_STEP = `${Job.BASE}/step2`;
            export const VALIDATION = `${BASE_STEP}/validation`;
            export const ID = `${BASE_STEP}/{id}`;
        }

        export namespace Step3 {
            const BASE_STEP = `${Job.BASE}/step3`;
            export const VALIDATION = `${BASE_STEP}/validation`;
            export const ID = `${BASE_STEP}/{id}`;
        }

        export namespace Step4 {
            const BASE_STEP = `${Job.BASE}/step4`;
            export const VALIDATION = `${BASE_STEP}/validation`;
            export const ID = `${BASE_STEP}/{id}`;
        }
    }

    export namespace Freelancer {
        export const BASE = 'freelancers';
        export const SEARCH = `${BASE}/search`;
        export const ID = `${BASE}/{id}`;
    }

    export namespace List {
        const BASE = 'list';
        export const MAJOR = `${BASE}/majors`;
        export const LANGUAGE = `${BASE}/languages`;
        export const SKILL = `${BASE}/skills`;
        export const ROLE = `${BASE}/roles`;
        export const PERMISSION = `${BASE}/permissions`;
    }

    export namespace Me {
        export const BASE = 'me';
        export const FULL_INFO = `${BASE}/full-info`;
        export const APPLIES = `${BASE}/applies`;
        export const REVIEWS = `${BASE}/reviews`;
        export const JOBS_IN_PROGRESS = `${BASE}/jobs-in-progress`;
        export const JOBS_COMPLETED = `${BASE}/jobs-completed`;
    }

    export namespace Home {
        const BASE = 'home';
        export const MAJOR = `${BASE}/top-10-most-reputable-freelancers`;
    }

    export namespace Employer {
        const BASE = 'employer';
        export const RECRUITING_JOBS = `${BASE}/recruiting-jobs`;
        export const INVITE_FREELANCER = `${BASE}/invite-freelancer/{freelancerId}`;
    }
}
